src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/resources/stylesheet.css

```
@@ -863,3 +863,24 @@ table.striped > tbody > tr > th {
         padding-right: 12px;
     }
 }
+
+pre.snippet {
+    background-color: #e8e8e8;
```

I think the plain gray clashes a bit with the javadoc gray scale which has a hint of blue in it. Something like the value below works better within that scale. 
```suggestion
    background-color: #ebecee;
```